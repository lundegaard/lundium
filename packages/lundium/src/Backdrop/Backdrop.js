import React from 'react';
import { cx } from 'ramda-extension';
import PropTypes from 'prop-types';

import Box from '../Box';

const Backdrop = ({ isVisible, className, ...rest }) => (
	<Box
		px={[0, 2]}
		py={[0, 1]}
		className={cx('backdrop', { ['show']: isVisible }, className)}
		{...rest}
	/>
);

Backdrop.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	/** Is backdrop visible */
	isVisible: PropTypes.bool,
	...Box.propTypes,
};

export default Backdrop;
