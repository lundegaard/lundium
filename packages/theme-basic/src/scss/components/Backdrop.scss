@import '../core';

/*
 -  Backdrop
 -  (<div class="backdrop" />)
 */

.backdrop {
	position: fixed;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	display: none;
	justify-content: center;
	align-items: center;
	z-index: $zindex-modal-backdrop;
	background-color: rgba($theme-gray-800, .3);
	opacity: 0;
	transition: opacity ( $transition-speed * 2 ) $transition-easing;

	// Media queries
	@include media-breakpoint-up(sm) {
		padding: ( $spacer * 2 );
	}

	&.show {
		display: flex;
		opacity: 1;

		.modal {
			transform: translateY(0);
		}
	}
}