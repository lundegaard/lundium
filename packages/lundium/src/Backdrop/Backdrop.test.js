import React from 'react';
import { shallow } from 'enzyme';

import Backdrop from './Backdrop';

describe('<Backdrop />', () => {
	const requiredProps = {};

	it('shallow', () => {
		const wrapper = shallow(
			<Backdrop {...requiredProps}>
				<div />
			</Backdrop>,
		);
		expect(wrapper).toMatchSnapshot();
	});
});
