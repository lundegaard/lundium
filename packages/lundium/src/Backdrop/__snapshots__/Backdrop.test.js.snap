// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> shallow 1`] = `
<forwardRef(Box)
  className="backdrop"
  px={
    Array [
      0,
      2,
    ]
  }
  py={
    Array [
      0,
      1,
    ]
  }
>
  <div />
</forwardRef(Box)>
`;
